### ISSUE TITLE
SCI score for SCER

### ISSUE THREAD

**Issue by seanmcilroy29** [Green-Software-Foundation/scer#23]
If kWh consumed is the true basis of the SCI score for SCER (since the energy mix in the deployment environment is unknown), should that be included in the label/rating? CO2e codes more information to the consumer and should be kept, but kWh could be relevant too.

**Comment by smol84** [comment]
I think KwH is a good inclusion. With C02e it is more opaque, does it include market or location based emissions? Jave they just offset rather than used carbon free energy etc...

Adding energy makes that more transparent. However won't include any scope 3 type missions (such as manufacture of kit that the softwate runs on)  which c02e would

**Comment by chrisxie-fw** [comment]
This is good catch, kWH may be a good data to be displayed in the labelling. For example, a LLMs trained by using nuclear power consumer a lot of kWH, but zero CO2e. In theory, these are all good info to be displayed in the labelling to give people a more comprehensive view of the rating, but in reality, for some particular applications, the real estate of the label may be constrained, and can't display that much information. So I guess this is a use-case specific decision that needs to be made at the application level.

**Comment by seanmcilroy29** [comment]
**From Charlene Wong (IMDA)**
As you might know, we have invested funds for companies in Singapore to put the green software patterns into trials to find out the impact. The trials are in progress. We also struggled in terms of the kWh vs CO2e. Reason is that we have companies who are on prem shifting to cloud and in this case, on-prem is typically on kWh and SG’s grid factor aren’t attractive due to the geographic limitations we are in.

**Comment by chrisxie-fw** [comment]
Yes, that's a perfect case in point that different use cases may call for different measurements. If a move may incur more CO2e's, then it may deserve a more in-depth understanding why it is so, and how to mitigate it. It may help uncover new opportunities for carbon efficiencies. This is captured in the [SCER framework/categorizations](https://github.com/Green-Software-Foundation/scer/blob/Dev/Software_Carbon_Efficiency_Rating/Software_Carbon_Efficiency_Rating_Specification.md#21-software-categorization-considerations), in the sense of comparing apples to apples. In the case of SG, if SG's grid factor aren't attractive due to geographic limitations as you said, maybe, that is a good finding to advise the government policy makers to put in maybe tax incentives to encourage investment and adoptions of technologies or standards such as the [Open Renewable Energy Systems (ORES)](https://github.com/open-renewable-energy-systems/ores) to address the carbon challenges.

**Comment by jawache** [comment]
Since the title of this issue is SCI score for SCER I think it's important to make clear that SCI is **Carbon per R**.  If you are measuring KwH per R or even Something per KwH those are **not** SCI scores, those are something else entirely, you can call them what you want, just not SCI scores :)

To answer your question @smol84 and @chrisxie-fw the SCI has very specific provisions about the use of offsets and market based instruments. The exact language they landed on was this:

> If the electricity consumption is connected to a grid, the short run marginal, long run marginal, or average emissions grid intensity of that grid shall be used

We spent almost 3 months discussing just that one sentence so there has been plenty of discussion and consensus! It was a hard 3 months :/ Ultimately what the above means is you can't use your purchases of renewable energy to offset, you have to actually report the carbon based on the actual carbon emissions from the actual energy you used at that moment in time.

**Comment by chrisxie-fw** [comment]
@jawache thank you for your comment and for providing context on using **Carbon per R** for SCI. I agree with your clarification that if SCI is used, it would be **Carbon per R**, not **kwh per R**. Does it provide value if someone displays both **Watt** and **SCI** information on their products and services, like this one, it's up to the individual/org to decide:

<img width="1040" alt="image" src="https://github.com/Green-Software-Foundation/scer/assets/54118227/d185ab9d-1ec1-4615-9c12-96ebb53848ac">

**Comment by jawache** [comment]
@chrisxie-fw it might add value and agree it's up to the org to decide.

Where I'm confused is that is SCER trying to align to standards like the SCI, or create new standards for KwH/R? In other threads it's been mentioned that the goal is to align to existing standards but as far as I'm aware there isn't a standard for KwH/R so we'd need to define that as a prerequisite (another spec, SEI, Software Energy Intensity?)

**Comment by smol84** [comment]
Whilst I agree with the use of marginal emissions per your example below. I think we may need to update the wprding inthe standard to outline where that information is gathered. Because on face value getting an organisation to understand their marginal emission from a power supplier is almost impossible, it can be estimated via algorithms, which is what electricity maps provides. So perhaps updating the owridng to list recommended sources? 

It may be worth simplifying this in the language of location and market based emissions - so that it's easier to include the example of why we explicitly are not considering market emission factors (because they can include ways to reduce emissions that are not immediately consequential). I think calling out why we use X instead of Y will ensure people don't use the wrong measure for emissions.

> If the electricity consumption is connected to a grid, the short run marginal, long run marginal, or average emissions grid intensity of that grid shall be used


Regarding Energy - I think there is value in having it as an option, perhaps you are right in needing something like SEI. We're certainly seeing examples of needing to compare impacts of energy consumption and not just emissions. Particularly considering IRENA / to hit existing climate goals we need to triple renewables and double energy efficiency (for us that is being able to pick more energy efficient software)

**Comment by jawache** [comment]
@smol84 so the specific language above took 3 months to negotiate (welcome to your future!) I just helped facilitate the conversations, I expressed no opinions. The discussions were deep and there were strong opinions held by multiple organisations which resulted in the above compromise. You're free to suggest your change though! If there is consensus agreement it will get changed, that's our rule. I would start with a discussion in the SCI repo, however I'm guessing the group doesn't particularly want to revisit this at the moment.

There is a separate sci guide project which has more helpful advice which is where that type of educational info would go, but the spec itself has to be concise. The team had to remove quite a lot of helpful how to use info to get it into a format acceptable to ISO. I'd imagine we need to move a lot of the SCER content to a guide also when the time comes and leave in there just the concise words needed to explain how to conform to the spec .

RE: SEI if it makes sense and adds value then that should be proposed as a separate specification project is the argument I'm making, so it can be discussed and approved on its own merits rather than an appendix to SCER. However after reflection I think you can get what you want with just the SCI by just fixing the I and M variables to constants, for the use case in benchmarking that should be sufficient and equivalent to just an energy measure.
